Sources/TuistKit/Commands/Cache/CacheOptions.swift

```diff
@@ -20,4 +21,12 @@ struct CacheOptions: ParsableArguments {
         help: "When passed it caches the targets for simulator and device using xcframeworks."
     )
     var xcframeworks: Bool = false
+
+    @Option(
+        name: .long,
+        help: "Output type of xcframeworks when --xcframeworks is passed (device/simulator)"
+    )
+    var xcframeworksType: CacheXCFrameworkType?
```

Comment:
Maybe this can be `--destination device` and `--destination simulator` instead of `--xcframeworks-type device/simulator`? Imho as a developer, you want to say "I want to cache xcframeworks for device" and don't think of it as a xcframework type. This way, it makes it easier to also use this option for non-xcframework builds if we ever decide to down the line.

Reply:
I wonder if it would also make the implementation easier and the output type wouldn't be an associated value but a separate boolean. I don't have a really strong opinion on that, though

Reply:
What do you mean by saying "a separate boolean"? I don't really understand. Do you mean like `--deviceDestination` and `--simulatorDestination` (a `@Flag`)?

Reply:
No, I meant in the internal representation (not a part of the command) we currently need to drill in the enum [here](https://github.com/tuist/tuist/pull/5075/files#diff-526ca7f5338080e6a6629b03317fd380c436eef1354a6104210436f63ee96b5dR69). Given this setting is not applied on a case-by-case basis, we can have a boolean to determine those `shouldBuildForSimulator` and `shouldBuildForDevice`. But as mentioned, we can just change the CLI parameter to `--destination` 👍